dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp

```diff
@@ -151,16 +160,18 @@ void CHBlockChunkCodecStream::encode(const Block & block, size_t start, size_t e
         const ColumnWithTypeAndName & column = block.safeGetByPosition(i);
 
         writeStringBinary(column.name, *output);
-        writeStringBinary(column.type->getName(), *output);
+        writeStringBinary(CodecUtils::convertDataTypeNameByMppVersion(column.type->getName(), mpp_version), *output);
 
         if (rows)
-            WriteColumnData(*column.type, column.column, *output, 0, 0);
+            CHBlockChunkCodec::WriteColumnData(*column.type, column.column, *output, 0, 0, mpp_version);
```

Comment:
What about getting a `ser_type` and use it for writing down the data type name and also writeColumnData?

```
auto ser_type = CodecUtils::convertDataTypeNameByMppVersion(column.type->getName(), mpp_version);

writeStringBinary(ser_type, *output);
if (rows)
    CHBlockChunkCodec::WriteColumnData(*ser_type, column.column, *output, 0, 0);
```